lib/publisher.js

```diff
@@ -45,31 +47,22 @@ class Publisher {
 
   /**
    * Publish a message
-   * @param {Buffer} message - A Node.js buffer object, representing the message to be published.
-   * @return {undefined}
+   * @param {Object} message - An object of the topic message.
+   * @return {void}
    */
   publish(message) {
-    let msg;
-    if (message instanceof Buffer) {
-      msg = message;
-    } else if (typeof message.getBuffer === 'function') {
-      msg = message.getBuffer();
-    }  // TODO(kenny): check messageType and convert raw values to a ros message
-
-    if (msg) {
-      rclnodejs.publishMessage(this._handle, msg);
+    let rawRosMessage = this._typeClass.getRefBuffer(message);
+    if (rawRosMessage) {
```

Comment:
I believe we still need to add an 'else if' branch to translate a plain JavaScript value/object to message if possible, this would be done by leverage the _typeClass.import() function (TODO: define such a function in dot template)
After it has been done, we can publish a string message using
```
publisher.publish('plain string');
```